mitodl/mit-learn
# show more button for v2 drawer dates

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5874

### Description (What does it do?)
This PR adds a "show more" link to the start dates section of the info displayed in the v2 learning resource drawer. This expands and collapses the start dates for the course.

### Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/b488049a-6e6a-420e-a8ed-bd828d11e4a0)
![image](https://github.com/user-attachments/assets/78ae23af-f782-45c4-a5d7-721baaf628d0)
![image](https://github.com/user-attachments/assets/2af1dea2-768a-40de-ba21-7a18473db678)
![image](https://github.com/user-attachments/assets/e38deb2f-3e42-451c-87ae-64377778d3cb)

### How can this be tested?
 - If you have Posthog set up locally, enable the `lr_drawer_v2` flag
 - If you don't have Posthog set up locally, you can force `drawerV2` to be `true` in `LearningResourceDrawer.tsx`
 - Spin up this branch of `mit-learn`
 - Ensure you have sufficient data backpopulated into your local instance, especially Sloan courses
 - Search for a course with identical data but with multiple start dates, like "Artificial Intelligence in Pharma and Biotech" for example
 - Confirm that you can click "Show more" to expand the rest of the start dates, that the link text changes to "Show less" after it expands, and that you can click "Show less" to collapse back down
 - Search for a course with different data and multiple start dates, such as "Leading the AI-Driven Organization"
 - Confirm that all of the start dates are shown without the show more button
